sw/device/lib/runtime/hart.c

```diff
@@ -0,0 +1,22 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/runtime/hart.h"
+
+#include <stdbool.h>
+
+#include "sw/device/lib/runtime/ibex.h"
+
+extern void wait_for_interrupt(void);
+
+void busy_sleep_micros(size_t microseconds) {
+  size_t cycles = kIbexClockFreqHz * microseconds / 10000000;
+  ibex_busy_loop(cycles);
+}
+
+noreturn void abort(void) {
```

Comment:
Stab in the dark here, but would there be some desired state that we want the system to go (specific actions to take) in case of a fatal error? Just guessing that it could be a practice that is taken in security systems, similar to safety critical systems - where in case of a critical error the system should transition to a well defined safe state (whatever this state was determined to be).

I suppose if this guess has any merit, it should be a part of a separate discussion anyway.

Reply:
Please raise this point as a separate issue, it's not really in scope for this PR (and will not reach the right people here).

Reply:
Agreed.  The state that the system ends up in will be different for different stages of boot, but that's something we need to determine elsewhere.

Reply:
I think we should table this, but I'll also add that it's unclear if `abort()` is even the right place for signaling "omg my world is one fire", especially since the compiler likes to emit synthetic `abort` calls.